tests/unit/test_lifecycle.py

```
@@ -0,0 +1,630 @@
+        key2 = _derive_fernet_key("same-secret")
+        assert key1 == key2
+
+    def test_derive_key_different_for_different_secrets(self):
+        key1 = _derive_fernet_key("secret-1")
+        key2 = _derive_fernet_key("secret-2")
+        assert key1 != key2
+
+    def test_encryption_secret_uses_sso_client_secret(self):
+        """Verify _get_encryption_secret reads SSO_CLIENT_SECRET from settings."""
+        mock_settings = MagicMock()
+        mock_settings.SSO_CLIENT_SECRET = "my-sso-secret"
```

Ruff (0.15.21)

[error] Possible hardcoded password assigned to: "SSO_CLIENT_SECRET"

(S105)